omas/omas_physics.py

```diff
@@ -93,6 +93,58 @@ def equilibrium_stored_energy(ods, update=True):
     return ods_n
 
 
+@add_to__ODS__
+@preprocess_ods('core_profiles', 'equilibrium')
+def calc_line_den(ods, R1, R2, Z1, Z2, line_grid=2000, time_index=0):
+    ods_n = ods
+    if not update:
+        from omas import ODS
+
+        ods_n = ODS().copy_attrs_from(ods)
+
```

Comment:
@jmcclena can you make it so that by default `time_index=None` and if so it calculates it for all time slices?

You can simply call your function recursively:

```suggestion
    if time_index is None:
        for time_index in range(len(ods['equilibrium']['time_slice'])):
            calc_line_den(ods_n, R1=R1, R2=R2, Z1=Z1, Z2=Z2, line_grid=line_grid, time_index=time_index, update=False)
        return ods_n
```
```

Reply:
> nice addition, I was trying to add this to tau_e_h98 and Greenwald. What are you giving to the following params:
> R1, R2, Z1, Z2 in the function def calc_line_den(ods, R1, R2, Z1, Z2) @jmcclena
@TimSlendebroek 
This function should work for any two points that pass through the plasma boundary. For DIII-D chords, it should be.

V1 chord: R1 =  R2 = 1.48 m, Z1=1.5, Z2=-1.5
V2 chord: R1 =  R2 = 1.94 m, Z1=1.5, Z2=-1.5
V3 chord: R1 =  R2 = 2.10 m, Z1=1.5, Z2=-1.5
R chord: Z1=Z2=0, R1 = 1.0 and R2 = 2.4

Reply:
@jmcclena you can add a DIII-D specific function in OMFIT under `OMFIT-source/omfit/classes/omfit_omas_d3d.py`
@eldond may be able to suggest how to best do this.

Also, could you also add a regression test that covers this new physics function?

Reply:
omfit_omas_d3d already has the interferometer geometry.